### PR TITLE
snap: make `snap login` ask for email if not given as argument

### PR DESCRIPTION
**Pull request by mvo5**
Addresses https://trello.com/c/V4pcInYj 

TestLoginAskEmail is slightly ugly, suggestions welcome!

### CODE REVIEW

File: cmd/snap/cmd_login.go [hunk 1]
```diff
@@ -99,15 +100,26 @@ func requestLogin(email string) error {
...
 
 func (x *cmdLogin) Execute(args []string) error {
 	if len(args) > 0 {
 		return ErrExtraArgs
 	}
 
-	err := requestLogin(x.Positional.Email)
+	email := x.Positional.Email
+	if email == "" {
+		fmt.Fprint(Stdout, "Please input your email address: ")
```
Comment (chipaca) [inline cmd/snap/cmd_login.go]: i18n.G?
Also, given the other prompts are just `<thing you should enter>`, maybe this should be just `"email address"`?

Comment (mvo5) [inline cmd/snap/cmd_login.go]: Makes sense, thanks!

**Comment by chipaca** [comment]
still no i18n tho :-)

File: cmd/snap/cmd_login.go [hunk 2]
```diff
@@ -99,15 +100,26 @@ func requestLogin(email string) error {
...
 
-	err := requestLogin(x.Positional.Email)
+	email := x.Positional.Email
+	if email == "" {
+		fmt.Fprint(Stdout, i18n.G("Email address: "))
+		in, _, err := bufio.NewReader(Stdin).ReadLine()
+		if err != nil {
+			return err
+		}
+		email = string(in)
```
Comment (zyga) [inline cmd/snap/cmd_login.go]: You could trim the string like you do with the input argument above.

File: cmd/snap/cmd_login_test.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+}
+
+func (s *SnapSuite) TestLoginAskEmail(c *C) {
+	n := 0
+	s.RedirectClientToTestServer(makeLoginTestServer(c, &n))
+
+	// send the email
+	fmt.Fprint(s.stdin, "foo@example.com\n")
+	// send the password
+	fmt.Fprint(s.term, "some-password\n")
```
Comment (zyga) [inline cmd/snap/cmd_login_test.go]: Oh, `s.term` is nice :-)

**Pull request merged**